src/client/finAuthFetch.ts

```diff
@@ -52,44 +55,71 @@ export async function finAuthFetch(
 			resource = await authenticator.preAuthenticate(resource, options)
 		}
 
-		const fetchMethod = options.fetch ?? defaultFetch
+		fetchMethod = options.fetch ?? defaultFetch
 
-		// Pipe request to the csrf fetch function
-		resp = await fetchMethod(resource, options)
+		// Pipe request to the fetch method
+		try {
+			resp = await fetchMethod(resource, options)
 
-		// Check if response is an authentication fault
-		if (!(await authenticator.isAuthenticated?.(resp))) {
-			const maxTries = authenticator.maxTries ?? 3
-			let authSuccessful = false
+			// Check if response is an authentication fault
+			if (!(await authenticator.isAuthenticated?.(resp))) {
+				throw new AuthenticationError(
+					new Error('Request not authenticated')
+				)
+			}
 
-			// Attempt to authenticate until we have reached the max try threshold
-			for (let i = 0; i < maxTries; i++) {
-				const result = await authenticator.authenticate(
+			// Response was authenticated, return response
+			return resp
+		} catch (error) {
+			if (error instanceof AuthenticationError) {
```

Comment:
Following the other comment, this would now check for an `HttpError` and whether it was a real authentication issue via `isAuthenticationError`. 

I would also recommend using a type guard function to ensure it's an `HttpError` instance. Tip: for type guards and JS in general, I try not to use `instanceof` and prefer checking for the existance of properties. I do this just in case there are accidently two versions of `haystack-nclient` loaded into a project (which unfortunately can easily happen in JS).

Reply:
I used a global symbol because these types are pretty simple right now. I wanted something a bit concrete. The global symbol will work across versions.